ipa-core/src/report/hybrid.rs

```diff
@@ -224,35 +363,132 @@ impl<BK, V> HybridReport<BK, V>
     BK: SharedValue,
     V: SharedValue,
+    Replicated<BK>: Serializable,
+    Replicated<V>: Serializable,
+    <Replicated<BK> as Serializable>::Size: Add<U16>,
+    <Replicated<V> as Serializable>::Size: Add<U16>,
+    <<Replicated<BK> as Serializable>::Size as Add<<Replicated<BA64> as Serializable>::Size>>:: Output: ArrayLength,
+    <<Replicated<V> as Serializable>::Size as Add<<Replicated<BA64> as Serializable>::Size>>:: Output: ArrayLength,
 {
+    /// # Panics
+    /// If report length does not fit in `u16`.
+    pub fn encrypted_len(&self) -> u16 {
+        match self {
+            HybridReport::Impression(impression_report) => {
+                impression_report.encrypted_len() +1
+            }
+            HybridReport::Conversion(conversion_report) => {
+                conversion_report.encrypted_len() +1
+            }
+        }
+    }
+
+    /// # Errors
+    /// If there is a problem encrypting the report.
+    pub fn delimited_encrypt_to<R: CryptoRng + RngCore, B: BufMut>(
+        &self,
+        key_id: KeyIdentifier,
+        key_registry: &impl PublicKeyRegistry,
+        info: &HybridInfo,
+        rng: &mut R,
+        out: &mut B,
+    ) -> Result<(), InvalidHybridReportError> {
+        match self {
+            HybridReport::Impression(impression_report) => match info {
+                HybridInfo::Impression(impression_info) =>{
+                    out.put_u16_le(self.encrypted_len());
+                    out.put_u8(1u8);
+                    impression_report.encrypt_to(key_id, key_registry, impression_info, rng, out)},
+                HybridInfo::Conversion(_) => {
+                    Err(InvalidHybridReportError::WrongInfoType("Impression"))
+                }
+            },
+            HybridReport::Conversion(conversion_report) => match info {
+                HybridInfo::Conversion(conversion_info) =>{
+                    out.put_u16_le(self.encrypted_len());
+                    out.put_u8(0u8);
+                    conversion_report.encrypt_to(key_id, key_registry, conversion_info, rng, out)},
+                HybridInfo::Impression(_) => {
+                    Err(InvalidHybridReportError::WrongInfoType("Conversion"))
+                }
+            },
+        }
+    }
+
     /// # Errors
     /// If there is a problem encrypting the report.
     pub fn encrypt<R: CryptoRng + RngCore>(
         &self,
-        _key_id: KeyIdentifier,
-        _key_registry: &impl PublicKeyRegistry,
-        _rng: &mut R,
-    ) -> Result<Vec<u8>, InvalidReportError> {
-        unimplemented!()
+        key_id: KeyIdentifier,
+        key_registry: &impl PublicKeyRegistry,
+        info: &HybridInfo,
+        rng: &mut R,
+    ) -> Result<Vec<u8>, InvalidHybridReportError> {
+        match self {
+            HybridReport::Impression(impression_report) => match info {
+                HybridInfo::Impression(impression_info) =>
+                    // Prepend a 1u8 byte to indicate this is an impression report
```

Comment:
I'd prefer if we use a type here as encoding 0 and 1 in more than one place can get out of sync quickly.